File: 151002/Saprin/RV/publicator/src/main/java/by/bsuir/publicator/controller/StickerController.java
package by.bsuir.publicator.controller;

import by.bsuir.publicator.dto.StickerRequestTo;
import by.bsuir.publicator.dto.StickerResponseTo;
import by.bsuir.publicator.exception.EntityNotFoundException;
import by.bsuir.publicator.service.sticker.IStickerService;
import jakarta.validation.Valid;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.*;

import java.math.BigInteger;
import java.util.List;

@RestController
@RequestMapping("/stickers")
public class StickerController {

    private final IStickerService stickerService;

    @Autowired
    public StickerController(IStickerService stickerService) {
        this.stickerService = stickerService;
    }

    @GetMapping()
    @ResponseStatus(HttpStatus.OK)
    public List<StickerResponseTo> getStickers() {
        return stickerService.getStickers();
    }

    @GetMapping("/{id}")
    @ResponseStatus(HttpStatus.OK)
    public StickerResponseTo getStickerById(@PathVariable BigInteger id) throws EntityNotFoundException {
        return stickerService.getStickerById(id);
    }

    @PostMapping()
    @ResponseStatus(HttpStatus.CREATED)
    public StickerResponseTo createSticker(@RequestBody @Valid StickerRequestTo sticker) {
        return stickerService.createSticker(sticker);
    }

    @PutMapping()
    @ResponseStatus(HttpStatus.OK)
    public StickerResponseTo updateSticker(@RequestBody @Valid StickerRequestTo sticker) throws EntityNotFoundException {
        return stickerService.updateSticker(sticker);
    }

    @DeleteMapping("/{id}")
    @ResponseStatus(HttpStatus.NO_CONTENT)
    public void deleteSticker(@PathVariable BigInteger id) throws EntityNotFoundException {
        stickerService.deleteSticker(id);
    }

}

File: 151004/Danilov/publisher/src/main/kotlin/com/danilovfa/controller/StickerController.kt
package com.danilovfa.controller

import com.danilovfa.model.sticker.StickerCreateRequestTo
import com.danilovfa.model.sticker.StickerUpdateRequestTo
import com.danilovfa.model.sticker.toResponse
import com.danilovfa.service.StickerService
import com.danilovfa.util.exception.respondError
import io.ktor.http.*
import io.ktor.server.application.*
import io.ktor.server.request.*
import io.ktor.server.response.*
import io.ktor.server.routing.*

fun Route.stickerController(service: StickerService) {
    post("/stickers") {
        try {
            val request = call.receive<StickerCreateRequestTo>()
            val post = service.createSticker(request.name).toResponse()
            call.respond(HttpStatusCode.Created, post)
        } catch (exception: Exception) {
            call.respondError(exception)
        }
    }

    get("/stickers/{id}") {
        try {
            val id = call.parameters["id"]?.toLongOrNull()
            val post = service.getStickerById(id).toResponse()
            call.respond(HttpStatusCode.OK, post)
        } catch (exception: Exception) {
            call.respondError(exception)
        }
    }

    get("/stickers") {
        try {
            val response = service.getStickers().map { it.toResponse() }
            call.respond(HttpStatusCode.OK, response)
        } catch (exception: Exception) {
            call.respondError(exception)
        }
    }

    put("/stickers") {
        try {
            val request = call.receive<StickerUpdateRequestTo>()
            val response = service.updateSticker(request.id, request.name).toResponse()
            call.respond(HttpStatusCode.OK, response)
        } catch (exception: Exception) {
            call.respondError(exception)
        }
    }

    delete("/stickers/{id}") {
        try {
            val id = call.parameters["id"]?.toLongOrNull()
            service.deleteSticker(id)
            call.respond(HttpStatusCode.NoContent)
        } catch (exception: Exception) {
            call.respondError(exception)
        }
    }
}